src/main/webapp/app/shared/participant-scores/participant-scores-average-table/participant-scores-average-table.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, Input } from '@angular/core';
+import { ParticipantScoreAverageDTO } from 'app/shared/participant-scores/participant-scores.service';
+
+@Component({
+    selector: 'jhi-participant-scores-average-table',
+    templateUrl: './participant-scores-average-table.component.html',
+})
+export class ParticipantScoresAverageTableComponent {
+    @Input()
+    participantAverageScores: ParticipantScoreAverageDTO[] = [];
+    @Input()
+    isLoading = false;
+
+    extractParticipantName = (participantScoreAverageDTO: ParticipantScoreAverageDTO) => {
+        if (participantScoreAverageDTO.userName) {
+            return `${participantScoreAverageDTO.userName}`;
+        } else {
+            return `${participantScoreAverageDTO.teamName}`;
+        }
```

Comment:
remove the else:

```suggestion
        return `${participantScoreAverageDTO.teamName}`;
```

When username is empty, is teamName always set? Otherwise we would need a default return value

Reply:
Yes either the team name or the userName are set